src/transformers/models/opt/modeling_opt.py

```diff
@@ -1060,6 +1096,14 @@ def prepare_inputs_for_generation(
 
             input_ids = input_ids[:, remove_prefix_length:]
 
+        if attention_mask is not None and position_ids is None:
+            position_ids = _attention_to_position_ids(attention_mask)
+            if past_key_values:
+                position_ids = position_ids[:, -input_ids.shape[1] :]
+
+                # This `clone` call is needed to avoid recapturing cuda graphs with `torch.compile`'s  `mode="reduce-overhead`, as otherwise the input `position_ids` would have various stride during the decoding. Here, simply using `.contiguous()` is not sufficient as in the batch size = 1 case, `position_ids` is already contiguous but with varying stride which retriggers a capture.
```

Comment:
comment is way to long let's split it on more lines!

Reply:
K